plugin-api/src/main/java/org/hyperledger/besu/plugin/services/PermissioningService.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.plugin.services;
+
+import org.hyperledger.besu.plugin.services.permissioning.NodeConnectionPermissioningProvider;
+import org.hyperledger.besu.plugin.services.permissioning.NodeMessagePermissioningProvider;
+
+public interface PermissioningService extends BesuPluginService {
+  void registerNodePermissioningProvider(NodeConnectionPermissioningProvider provider);
+
+  void registerNodeMessagePermissioningProvider(NodeMessagePermissioningProvider provider);
```

Comment:
what about `unregister`? it should be there as plugins might want to deregister too.

Moreover, I see some naming convention of other services where is used the convention `addListener`/`removeListener`. shouldn't we use also here add/remove?

Reply:
The listeners are different, you're observing something that has happened. You are not changing the behaviour of the node. You may want to listen to blocks, then after block number n listen to events for 100 blocks or something. Either way, the act of observing doesn't change the behaviour of the node. 

Whereas with things that change behaviour I don't it make sense, would you run a node with permissioning enabled and then turn permissioning off whilst it's running. Maybe you would, but you would need to know information about the block height when making that decision.

Reply:
non-blocking:

you allow also multiple providers. shouldn't be only 1? as if I follow your thoughts, then one provider can affect other one, so there can be still issues. 

plus, as you have multiple providers, might always good to unregister them even for memory management (except if we were using WeakReferences which will be cleared up automatically).